src/OpenSolid/QuadraticSpline2d.elm

```diff
@@ -460,59 +461,78 @@ splitAt t spline =
     )
 
 
-{-| A parameterization of a quadratic spline by arc length.
--}
-type ArcLengthParameterization
-    = ArcLengthParameterization QuadraticSpline2d ArcLengthParameterization.ArcLengthParameterization
-
-
 {-| Build an arc length parameterization of the given spline.
 -}
-arcLengthParameterization : Float -> QuadraticSpline2d -> ArcLengthParameterization
-arcLengthParameterization tolerance spline =
-    ArcLengthParameterization spline <|
-        ArcLengthParameterization.build parameterizationConfig tolerance spline
+arcLengthParameterized : Float -> QuadraticSpline2d -> ArcLengthParameterized QuadraticSpline2d
+arcLengthParameterized tolerance spline =
+    ( spline, ArcLength.parameterization arcLengthConfig tolerance spline )
 
 
-parameterizationConfig : ArcLengthParameterization.Config QuadraticSpline2d
-parameterizationConfig =
+arcLengthConfig : ArcLength.Config QuadraticSpline2d
+arcLengthConfig =
     { bisect = bisect
-    , lengthBounds = lengthBounds
+    , speed = speed
     }
 
 
-lengthBounds : QuadraticSpline2d -> ( Float, Float )
-lengthBounds spline =
+speed : QuadraticSpline2d -> Interval
+speed spline =
     let
-        ( p1, p2, p3 ) =
-            controlPoints spline
+        vA =
+            startDerivative spline
+
+        vB =
+            endDerivative spline
+
+        vC =
+            Vector2d.difference vA vB
+
+        aSquared =
+            Vector2d.squaredLength vA
+
+        bSquared =
+            Vector2d.squaredLength vB
+
+        cSquared =
+            Vector2d.squaredLength vC
+
+        a =
+            sqrt aSquared
 
-        lowerBound =
-            Point2d.distanceFrom p1 p3
+        b =
+            sqrt bSquared
 
-        upperBound =
-            Point2d.distanceFrom p1 p2 + Point2d.distanceFrom p2 p3
+        minValue =
+            if cSquared == 0 then
```

Comment:
Is this `speed` the norm of the derivative ? if so, I think in case `vC == 0` it means it is a round trip with same starting and ending points? And in that case the derivative nullifies at half course right? Should it be 0 then?

Reply:
Oops, ok `vC = vA - vB` not `+` so it is not the round trip but the degenerate segment. Got it!